### PR TITLE
Implementation rotatesArrayOfCoordinates. Issue #9

### PR DESCRIPTION
**Pull request by PashaBeshchuk**

### CODE REVIEW

**Comment by zbstof** [comment]
Realization => Implementation

File: tetris.js [hunk 1]
```diff
@@ -308,5 +308,6 @@ module.exports = {
 		let coordinateDifference_X = coordinatesPivot.y - coordinatesElement.y
 		let coordinateDifference_Y = coordinatesPivot.x - coordinatesElement.x
 		return { x: coordinatesPivot.x - coordinateDifference_X, y: coordinatesPivot.y + coordinateDifference_Y }
-	}
+	},
+	rotatesArrayOfCoordinates: function(){}
```
Comment (zbstof) [inline tetris.js]: rotateArrayOfCoordinates
All methods should be in Present tense

File: tetris.js [hunk 2]
```diff
@@ -309,5 +309,14 @@ module.exports = {
 		let coordinateDifference_Y = coordinatesPivot.x - coordinatesElement.x
 		return { x: coordinatesPivot.x - coordinateDifference_X, y: coordinatesPivot.y + coordinateDifference_Y }
 	},
-	rotatesArrayOfCoordinates: function(){}
+	rotatesArrayOfCoordinates: function(arrayCoordinates, coordinatePivot){
+		let localArrayCoordinates = []
+		if(arrayCoordinates.length === 0){
```
Comment (zbstof) [inline tetris.js]: Is this check necessary?

File: tetris.js [hunk 3]
```diff
@@ -308,5 +308,15 @@ module.exports = {
...
 		let coordinateDifference_Y = coordinatesPivot.x - coordinatesElement.x
 		return { x: coordinatesPivot.x - coordinateDifference_X, y: coordinatesPivot.y + coordinateDifference_Y }
+	},
+	rotatesArrayOfCoordinates: function(arrayCoordinates, coordinatePivot){
+		let localArrayCoordinates = []
+		if(arrayCoordinates.length === 0){
+			return arrayCoordinates
+		}	
+		for(let arrayElement of arrayCoordinates){
+			localArrayCoordinates.push(this.rotateElement(arrayElement,coordinatePivot[0]))
```
Comment (zbstof) [inline tetris.js]: Pivot should be a single element, and not array

File: tetris.js [hunk 4]
```diff
@@ -317,6 +303,44 @@ module.exports = {
 		for(let arrayElement of arrayCoordinates){
 			localArrayCoordinates.push(this.rotateElement(arrayElement,coordinatePivot[0]))	
 		}
-		return localArrayCoordinates
+		return this.sort(localArrayCoordinates)
+	},
+	sort: function(arrayOfCoordinates){
```
Comment (zbstof) [inline tetris.js]: 1. Your imlementation is very complex and probably has a lot of bugs
2. Javascript already has array sort method. Did you google before writing your own? This method should be single line of code

Comment (zbstof) [reply tetris.js]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#Syntax

File: tetris.js (hunk 2 above)
Comment (PashaBeshchuk) [reply tetris.js]: I think yes
If array === 0  then it is empty

File: tetris.js [hunk 5]
```diff
@@ -277,7 +277,13 @@ module.exports = {
 			}
 
 		}
-		return this.sort(arrayOfCoordinates);
+		return arrayOfCoordinates.sort(function(firstCoordinate,secondCoordinate){
+			let coordinates_X = firstCoordinate.x - secondCoordinate.x;
+			let coordinates_Y = firstCoordinate.y - secondCoordinate.y;
+			return coordinates_X,coordinates_Y;
```
Comment (zbstof) [inline tetris.js]: Does this work? I thought function should return number

File: tetris.js (hunk 2 above)
Comment (zbstof) [reply tetris.js]: Ok, why is in necessary?

File: tetris.js (hunk 3 above)
Comment (PashaBeshchuk) [reply tetris.js]: Functiion getCoordinates return array, because it is an array

File: tetris.js (hunk 2 above)
Comment (zbstof) [reply tetris.js]: This check is not needed, remove

File: tetris.js (hunk 3 above)
Comment (zbstof) [reply tetris.js]: This is not according to requirements, change parameters of rotatesArrayOfCoordinates

File: tetris.js (hunk 2 above)
Comment (PashaBeshchuk) [reply tetris.js]: Done

**Comment by zbstof** [comment]
Please write Done under comments after you've updated PR, and not before that.
Because usually it is self-evident what have you updated, and a separate comment is not necessary

File: tetris.js (hunk 5 above)
Comment (PashaBeshchuk) [reply tetris.js]: not exactly yet,two tests fail
just discovered

File: tetris.js [hunk 6]
```diff
@@ -279,8 +279,7 @@ module.exports = {
 		}
 		return arrayOfCoordinates.sort(function(firstCoordinate,secondCoordinate){
 			let coordinates_X = firstCoordinate.x - secondCoordinate.x;
-			let coordinates_Y = firstCoordinate.y - secondCoordinate.y;
-			return coordinates_X,coordinates_Y;
+			return coordinates_X;
```
Comment (zbstof) [inline tetris.js]: You are not sorting by y

File: tetris.js [hunk 7]
```diff
@@ -308,5 +299,21 @@ module.exports = {
 		let coordinateDifference_X = coordinatesPivot.y - coordinatesElement.y
 		let coordinateDifference_Y = coordinatesPivot.x - coordinatesElement.x
 		return { x: coordinatesPivot.x - coordinateDifference_X, y: coordinatesPivot.y + coordinateDifference_Y }
+	},
+	rotatesArrayOfCoordinates: function(arrayCoordinates, coordinatePivot){
+		let localArrayCoordinates = []
+		let localCoordinatePivot = coordinatePivot[0]
```
Comment (zbstof) [inline tetris.js]: No, function should accept element, not array

File: tetris.js [hunk 8]
```diff
@@ -308,5 +299,21 @@ module.exports = {
...
+		for(let arrayElement of arrayCoordinates){
+			localArrayCoordinates.push(this.rotateElement(arrayElement,localCoordinatePivot))	
+		}
+		return localArrayCoordinates.sort(function(firstCoordinate,secondCoordinate){
+			let coordinates_X = firstCoordinate.x - secondCoordinate.x;
+			if(firstCoordinate.x === secondCoordinate.x){
+				let coordinates_Y = firstCoordinate.y - secondCoordinate.y;
+				return coordinates_X, coordinates_Y;
```
Comment (zbstof) [inline tetris.js]: Again, how does this work? You should always return number

File: tetris.js [hunk 9]
```diff
@@ -308,5 +299,21 @@ module.exports = {
...
 		let coordinateDifference_Y = coordinatesPivot.x - coordinatesElement.x
 		return { x: coordinatesPivot.x - coordinateDifference_X, y: coordinatesPivot.y + coordinateDifference_Y }
+	},
+	rotatesArrayOfCoordinates: function(arrayCoordinates, coordinatePivot){
+		let localArrayCoordinates = []
+		let localCoordinatePivot = coordinatePivot[0]
+		for(let arrayElement of arrayCoordinates){
+			localArrayCoordinates.push(this.rotateElement(arrayElement,localCoordinatePivot))	
+		}
+		return localArrayCoordinates.sort(function(firstCoordinate,secondCoordinate){
```
Comment (zbstof) [inline tetris.js]: Why are you sorting fifferently here and on line 282?

File: tetris.js (hunk 8 above)
Comment (PashaBeshchuk) [reply tetris.js]: it checks the elements of the array and sorts them

Comment (zbstof) [reply tetris.js]: Ok, if you are so confident, then create separate PR with sorting function and tests for it

Comment (zbstof) [reply tetris.js]: `return coordinates_X, coordinates_Y;`
is the same as 
`return coordinates_Y;`
in this context, please change

File: tetris.js [hunk 10]
```diff
@@ -308,5 +303,20 @@ module.exports = {
...
 		return { x: coordinatesPivot.x - coordinateDifference_X, y: coordinatesPivot.y + coordinateDifference_Y }
+	},
+	rotatesArrayOfCoordinates: function(arrayCoordinates, coordinatePivot){
+		let localArrayCoordinates = []
+		for(let arrayElement of arrayCoordinates){
+			localArrayCoordinates.push(this.rotateElement(arrayElement,coordinatePivot))	
+		}
+		return localArrayCoordinates.sort(function(firstCoordinate,secondCoordinate){
+			let coordinates_X = firstCoordinate.x - secondCoordinate.x;
+			if(firstCoordinate.x === secondCoordinate.x){
```
Comment (zbstof) [inline tetris.js]: This is the same as `if (coordinates_X === 0) {`
please change

File: tetris.js [hunk 11]
```diff
@@ -308,5 +303,20 @@ module.exports = {
...
+				let coordinates_Y = firstCoordinate.y - secondCoordinate.y;
```
Comment (zbstof) [inline tetris.js]: it's more like `yCoordinateDifference`

File: tetris.js [hunk 12]
```diff
@@ -308,5 +303,20 @@ module.exports = {
...
 		let coordinateDifference_Y = coordinatesPivot.x - coordinatesElement.x
 		return { x: coordinatesPivot.x - coordinateDifference_X, y: coordinatesPivot.y + coordinateDifference_Y }
+	},
+	rotatesArrayOfCoordinates: function(arrayCoordinates, coordinatePivot){
+		let localArrayCoordinates = []
+		for(let arrayElement of arrayCoordinates){
+			localArrayCoordinates.push(this.rotateElement(arrayElement,coordinatePivot))	
+		}
+		return localArrayCoordinates.sort(function(firstCoordinate,secondCoordinate){
+			let coordinates_X = firstCoordinate.x - secondCoordinate.x;
```
Comment (zbstof) [inline tetris.js]: more like `xCoordinateDifference`

File: tetris.js (hunk 10 above)
Comment (PashaBeshchuk) [reply tetris.js]: without this, tests don't work

Comment (zbstof) [reply tetris.js]: 1. Why they don't work?
2. Did i suggest to remove this or replace this check? 
3. How firstCoordinate.x === secondCoordinate.x is different from coordinates_X === 0?

Comment (PashaBeshchuk) [reply tetris.js]: here I compare two numbers, but your version is not clear to me

Comment (PashaBeshchuk) [reply tetris.js]: 1. Я делаю сортировку по Y  только в том случае если Х равны, если делать постоянно такую проверку, то возвращаться будет не всегда корректные данные.
2  Я сделал замену на  from coordinates_X === 0 и так тесты не проходят
3 Как по мне тут выполняется не та операция что мне нужна. Мне нужно сравнить теперешний Х координату со следующей Х координатой

Comment (zbstof) [reply tetris.js]: 1. Exact error please
2. Exact error please
3. 
```
let coordinates_X = firstCoordinate.x - secondCoordinate.x;
firstCoordinate.x === secondCoordinate.x
===>
let coordinates_X = firstCoordinate.x - secondCoordinate.x;
firstCoordinate.x - secondCoordinate.x === 0
===>
let coordinates_X = firstCoordinate.x - secondCoordinate.x;
coordinates_X === 0
```

Comment (PashaBeshchuk) [reply tetris.js]: А чем этот вариант хуже моего?

**Pull request merged**